### PR TITLE
Mk 1805 document how taquito is tested

### LINKED ISSUE #1805: Documentation page for Taquito about how testing is done
Add to the Taquito docs a page that discusses how software testing for Taquito is done, listing the types of testing, how they are done, the frequency, concerns, philosophy etc.

unit
integration
test coverage
mutation testing
etc

### PR DESCRIPTION
**Pull request by michaelkernaghan**
close #1805

A Doc file for how Taquito is tested

### CODE REVIEW

File: docs/testing_strategies.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+---
+title: Taquito Testing Strategies
+id: testing_strategies
+author: Michael Kernaghan
+---
+ 
+### How Taquito is Tested
+ 
+- General Philosophy
```
Comment (dsawali) [inline docs/testing_strategies.md:9]: could we try and use H4 instead here? 

e.g. 
'#### How Taquito is Tested'

File: docs/testing_strategies.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ 
+We write and maintain tests so that we may check that the software we create will do what we think it should do and alert us when it doesn’t.
+This link offers more detail on software testing philosophy in general: https://www.codesimplicity.com/post/the-philosophy-of-testing/.
+ 
+- Assumptions
```
Comment (dsawali) [inline docs/testing_strategies.md:14]: here as well

File: docs/testing_strategies.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ 
+### Performance 
+
+Ecad DevOps maintains an extensive performance tracking monitoring setup using Loki and Grafana, which generates alerts when specific performance parameters are out of band.
+ 
+## Managing Tezos Protocol Migrations with Test Nets
+
+Each time Tezos changes protocol, there is a new test net, and old ones are deprecated. Contracts originated in a more senior test net must be originated again on the new testnet. We have to update RPC content values and recreate Live Code Example contracts. So each protocol change requires an overhaul of some of the test assets to suit the new protocol.
+
+The Taquito test suite will run tests in CICD against the current test net and the next coming test net. There is also testing of “Mondaynet,” which represents the bleeding edge of the available Tezos test code.
```
Comment (dsawali) [inline docs/testing_strategies.md:102]: We could also mention Dailynet as well

File: docs/testing_strategies.md (hunk 1 above)
Comment (michaelkernaghan) [reply docs/testing_strategies.md:9]: changed to 
#### General Philosophy

File: docs/testing_strategies.md (hunk 2 above)
Comment (michaelkernaghan) [reply docs/testing_strategies.md:14]: changed to 
####  Assumptions

File: docs/testing_strategies.md (hunk 3 above)
Comment (michaelkernaghan) [reply docs/testing_strategies.md]: added dailynet

File: docs/testing_strategies.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+ 
+We do Code Reviews whenever a developer seeks to merge code. Other team members review it for correctness, accuracy, conformance with Taquito design overall, and suitability. This process will rapidly find problems that testing would either miss or take wasteful cycles to resolve.  We will not merge code changes or new features unless they have been code reviewed and all requested changes are determined.
+ 
+### Static Code Analysis
+ 
+Static Code Analysis is run during the CICD cycle to do syntactic checks for errors in the code. Often a line marking a merge conflict or a violation of a coding format will cause a static analyzer to complain. During a CICD run, a Pull Request will be examined by [CodeQL](https://codeql.github.com/) and [Snyk](https://snyk.io/).
+ 
+### End-to-End Tests
+ 
+Taquito uses the Taquito Test Dapp and the Live Code examples in the documentation as end-to-end tests. The tests exercise the entire software stack between the blockchain node and the user-facing interface.  These tests show that all the components are working together. AT each Taquito release, these tests are checked, and the results are included in the release.
```
Comment (roxaneletourneau) [inline docs/testing_strategies.md:75]: small typo: AT -> at

**Review by roxaneletourneau — COMMENTED** [review]
Nice work :)

File: docs/testing_strategies.md (hunk 4 above)
Comment (michaelkernaghan) [reply docs/testing_strategies.md]: Thanks. fixed

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -57,3 +56,30 @@ jobs:
...
+  integration-tests-flextesa:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v2
+      with:
+        node-version: 12.x
+    - run: npm ci
+    - run: npm run lerna -- bootstrap
```
Comment (egarson) [inline .github/workflows/main.yml:69]: We should use the npm interface everywhere, so that changes to how it works are centralized

Comment (egarson) [reply .github/workflows/main.yml:69]: (Does this environment use package.json?)

**Review by egarson — APPROVED** [review]
The testing document is crystal clear and really well written! Just the right amount of information.

File: .github/workflows/main.yml (hunk 5 above)
Comment (michaelkernaghan) [reply .github/workflows/main.yml:69]: Are you asking for a change in the doc?
